pkg/catalog/manager/traverse.go

```diff
@@ -199,6 +197,11 @@ func (m *Manager) traverseAndUpdate(helm *helmlib.Helm, commit string, cmt *Cata
 		if catalog == nil || catalog.Name == "" {
 			return errors.New("Catalog is no longer available")
 		}
+		if commit == catalog.Status.Commit && v3.CatalogConditionUpgraded.IsTrue(catalog) {
```

Comment:
I think this is already done in https://github.com/rancher/rancher/blob/f8a2e3830b3a80afc9cd0f3691168a043492f8f1/pkg/catalog/manager/catalog_sync.go#L42-L52 to avoid update if commit is the same. Is there anything special to add this check?

Reply:
We reload the catalog in traverseAndUpdate because of the time required to perform the operation, and it appears sync may have queued up events before we have successfully marked the catalog as processed. Since traverseAndUpdate may have stale catalog data we should verify the catalog has not been processed after refreshing to end updating as soon as possible.